core/vm/evmc.go

```diff
@@ -222,28 +460,34 @@ func (host *hostContext) GetCode(addr evmc.Address) []byte {
 	return host.env.StateDB.GetCode(common.Address(addr))
 }
 
-func (host *hostContext) Selfdestruct(evmcAddr evmc.Address, evmcBeneficiary evmc.Address) {
+func (host *hostContext) Selfdestruct(evmcAddr evmc.Address, evmcBeneficiary evmc.Address) bool {
 	addr := common.Address(evmcAddr)
 	beneficiary := common.Address(evmcBeneficiary)
 	db := host.env.StateDB
 	if !db.HasSuicided(addr) {
 		db.AddRefund(vars.SelfdestructRefundGas)
 	}
 	db.AddBalance(beneficiary, db.GetBalance(addr))
-	db.Suicide(addr)
+	return db.Suicide(addr)
 }
 
 func (host *hostContext) GetTxContext() evmc.TxContext {
-	return evmc.TxContext{
-		GasPrice:   evmc.Hash(common.BigToHash(host.env.GasPrice)),
-		Origin:     evmc.Address(host.env.TxContext.Origin),
-		Coinbase:   evmc.Address(host.env.Context.Coinbase),
-		Number:     host.env.Context.BlockNumber.Int64(),
-		Timestamp:  host.env.Context.Time.Int64(),
-		GasLimit:   int64(host.env.Context.GasLimit),
-		Difficulty: evmc.Hash(common.BigToHash(host.env.Context.Difficulty)),
-		ChainID:    evmc.Hash(common.BigToHash(host.env.chainConfig.GetChainID())),
+	txCtx := evmc.TxContext{
+		GasPrice:  evmc.Hash(common.BigToHash(host.env.GasPrice)),
+		Origin:    evmc.Address(host.env.TxContext.Origin),
+		Coinbase:  evmc.Address(host.env.Context.Coinbase),
+		Number:    host.env.Context.BlockNumber.Int64(),
+		Timestamp: host.env.Context.Time.Int64(),
+		GasLimit:  int64(host.env.Context.GasLimit),
+		ChainID:   evmc.Hash(common.BigToHash(host.env.chainConfig.GetChainID())),
```

Comment:
`Difficulty:` ... ?  (This field gets repurposed with PoS; it needs to be handled carefully.)